src/tests/ftest/control/config_generate_output.py

```diff
@@ -0,0 +1,445 @@
+        address set.
+        4. Get fabric_iface and verify that it's in the interface set.
+        5. Repeat for all engines.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=control,config_generate_entries,basic_config
+        """
+        # Get necessary storage and network info.
+        self.prepare_expected_data()
+
+        # Call dmg config generate.
+        generated_yaml = self.get_dmg_command().config_generate(
+            access_points="wolf-a")
+
+        errors = []
+
+        # Iterate engines and verify scm_list, bdev_list, and fabric_iface.
+        engines = generated_yaml["engines"]
+        for engine in engines:
+            # Verify the scm_list value is in the SCM Namespace set. e.g.,
+            # if the value is /dev/pmem0, check pmem0 is in the set.
+            scm_list = engine["scm_list"]
+            for scm_dev in scm_list:
+                device_name = scm_dev.split("/")[-1]
+                if device_name not in self.scm_namespace_set:
+                    errors.append(
+                        "Cannot find SCM device name {} in expected set {}"\
+                            .format(device_name, self.scm_namespace_set))
+
+            # Verify the bdev_list values are in the NVMe PCI address set.
+            bdev_list = engine["bdev_list"]
+            for pci_addr in bdev_list:
+                if pci_addr not in self.pci_address_set:
+                    errors.append(
+                        "Cannot find PCI address {} in expected set {}".format(
+                            pci_addr, self.pci_address_set))
+
+            # Verify fabric interface values are in the interface set.
+            fabric_iface = engine["fabric_iface"]
+            if fabric_iface not in self.interface_set:
+                errors.append(
+                    "Cannot find fabric interface {} in expected set {}".format(
+                        fabric_iface, self.interface_set))
+
+        self.check_errors(errors)
+
+    def test_access_points(self):
+        """Test --access-points.
+
+        1. Call dmg config generate --access-points=wolf-a. Should pass.
+        2. Call dmg config generate --access-points=wolf-a:12345. Should pass.
+        3. Call dmg config generate --access-points=wolf-a:abcd. Should fail
+        because the port number is invalid.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,small
+        :avocado: tags=control,config_generate_entries,access_points
+        """
+        errors = []
+        dmg = self.get_dmg_command()
+
+        # One host.
+        expected_ap = "wolf-a"
+        generated_yaml = dmg.config_generate(access_points=expected_ap)
+        actual_ap = generated_yaml["access_points"][0].split(":")[0]
+
+        if expected_ap != actual_ap:
+            msg = "One host failed! Expected = {}; Actual = {}".format(
+                expected_ap, actual_ap)
+            errors.append(msg)
```

Comment:
This should be a function to test multiple cases:

- single host w/o a valid port
- single host w/ a valid port
- multiple host w/o a valid port
- multiple host w/ a valid port
- single host w/ an invalid port
- one host w/ a valid port and another host w/ an invalid port
- same host repeated twice

```
        def verify_access_point(self, input, failure_expected):
            errors = []
            check = {"input": input}
            check["expected"] = check["input"].split(",")
            if ":" not in check["input"]::
                check["expected"] = ["{}:10001".format(host) for host in check["expected"]]
            try:
                generated_yaml = dmg.config_generate(access_points=expected_ap)
                check["actual"] = generated_yaml["access_points"]
                if failure_expected:
                    errors.append("Failure was expected")
                elif sorted(check["expected"]) != sorted(check["actual"]):
                    errors.append("Unexpected access point: {} != {}".format(check["expected"], check["actual"]))
            except CommandFailure as err:
                if not failure_expected:
                    errors.append("Failure was not expected")
            
            return errors
```